code/modules/chemistry/chemicompiler_core.dm

```diff
@@ -830,23 +830,25 @@
 	var/obj/item/reagent_containers/holder = reservoirs[rid]
 	var/datum/reagents/R = holder.reagents
 	var/heating_in_progress = 1
-	//while(R.total_volume && heating_in_progress)
 
 	//heater settings
-	var/h_exposed_volume = 10
-	var/h_divisor = 10
 	var/h_change_cap = 25
+	var/heater_temp
+	var/difference = temp- R.total_temperature
 
-	var/element_temp = R.total_temperature < temp ? 9000 : 0												//Sidewinder7: Smart heating system. Allows the CC to heat at full power for more of the duration, and prevents reheating of reacted elements.
-	var/max_temp_change = abs(R.total_temperature - temp)
-	var/next_temp_change = clamp((abs(R.total_temperature - element_temp) / h_divisor), 1, h_change_cap)	// Formula used by temperature_reagents() to determine how much to change the temp
-	if(next_temp_change >= max_temp_change)																	// Check if this tick will cause the temperature to overshoot if heated/cooled at full power. Use >= to prevent reheating in the case the values line up perfectly
-		element_temp = (((R.total_temperature - (R.total_temperature-temp)*h_divisor) * (R.total_volume+h_exposed_volume)) - (R.total_temperature*R.total_volume))/h_exposed_volume
-		heating_in_progress = 0
+	if (difference >= 0)
+		heater_temp = R.total_temperature+ 2*min(difference,h_change_cap)
+	else
+		heater_temp = R.total_temperature+ 2*max(difference,-h_change_cap)
 
-	h_divisor = 35/h_divisor*100 // quick hack because idk wtf is going on here - Emily
+	if(heater_temp <= 0)
+		//we can't set the heater below 0, so temp lowers slower
+		if (abs(difference) <= 1)
+			heating_in_progress = 0
+	else if(abs(difference) <= h_change_cap)
+		heating_in_progress = 0
 
-	R.temperature_reagents(element_temp, h_exposed_volume, h_divisor, h_change_cap)
+	R.temperature_reagents(heater_temp, R.total_volume*100, R.composite_heat_capacity, h_change_cap)
```

Comment:
Looks like `set_reagent_temp` calls `temperature_reagents` on its own, and is also way simpler than... whatever `temperature_reagents` is doing (seriously that proc looks like a mess)
Would rather use that instead of this arbitrary multiply-by-100 thing